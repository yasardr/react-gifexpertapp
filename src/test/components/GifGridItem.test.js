import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Probando GifGridItem', () => { 
    const title = 'Un título';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );

    test('debe mostrar el componente <GifGridItem /> correctamente', () => { 
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de tener un párrafo con el title', () => { 
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    });

    test('debe de tener la imagen igual al url y alt de los props', () => { 
        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    });

    test('debe de tener animate_fadeIn', () => { 
        const div = wrapper.find('div');
        // const className = div.prop('className');
        // expect( className.includes('animate__fadeIn') ).toBe( true );
        expect( div.hasClass('animate__fadeIn') ).toBe( true );
    });
});