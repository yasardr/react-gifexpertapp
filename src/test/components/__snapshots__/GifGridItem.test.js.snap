// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando GifGridItem debe mostrar el componente <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
